Mage/Command/Factory.php

```diff
@@ -43,14 +43,17 @@ public static function get($commandName, Config $config)
             // try a custom command
             $className = 'Command\\' . $commandName;
 
+            // TODO use a custom exception
             if (!class_exists($className)) {
                 throw new Exception('Command "' . $commandName . '" not found.');
             }
         }
 
         /** @var AbstractCommand $instance */
+        // TODO dependencies like $config should be injected into constructor
```

Comment:
I don't want to introduce ServiceLocator or other external libs, but the constructor is a common place to inject dependencies that are mandatory. Right now, the configuration is injected with a setter even when the Command doesn't need it. It means there is a setter which isn't needed in every Command class nor does every Command has a dependency to extend `AbstractCommand`.

The suggested todos are future talk, because injecting only the needed dependencies would result into changing the factory.
If there are already some rules about what this project doesn't want to introduce, a text-information like a contribution.md would be nice so that devs creating a PR can follow that guideline. Since we already introduced unit tests together, one contribution rule could be, that a PR must contain unit tests for the new code.

BTW: I'm glad the todo comments were noticed by someone.